tests/ignite/metrics/test_functional.py

```diff
@@ -0,0 +1,78 @@
+import torch
+from ignite.engine import Engine
+from ignite.metrics import Metric, MetricsLambda, Precision, Recall
+import ignite.metrics.functional as F
+from sklearn.metrics import precision_score, recall_score, f1_score
+import numpy as np
+from pytest import approx
+
+
+def test_helper_ops():
+
+    def _test(composed_metric, metric_name, compute_true_value_fn):
+
+        metrics = {
+            metric_name: composed_metric,
+        }
+
+        y_pred = torch.rand(15, 10, 5).float()
+        y = torch.randint(0, 5, size=(15, 10)).long()
+
+        def update_fn(engine, batch):
+            y_pred, y = batch
+            return y_pred, y
+
+        validator = Engine(update_fn)
+
+        for name, metric in metrics.items():
+            metric.attach(validator, name)
+
+        def data(y_pred, y):
+            for i in range(y_pred.shape[0]):
+                yield (y_pred[i], y[i])
+
+        d = data(y_pred, y)
+        state = validator.run(d, max_epochs=1)
+
+        assert set(state.metrics.keys()) == set([metric_name, ])
+        np_y_pred = np.argmax(y_pred.numpy(), axis=-1).ravel()
+        np_y = y.numpy().ravel()
+        assert state.metrics[metric_name] == approx(compute_true_value_fn(np_y_pred, np_y))
+
+    precision_1 = Precision(average=False)
+    precision_2 = Precision(average=False)
+    mean_summed_precision1 = F.mean(precision_1 + precision_2).item()
+    mean_summed_precision2 = (precision_1 + precision_2).mean().item()
```

Comment:
@vfdev-5 Unfortunately the API here are slightly different from those on https://github.com/pytorch/ignite/pull/417